### PR TITLE
CI : improve compile running time #217

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
We want to reduce OSW all-model compile running time.
See #217

### CODE REVIEW

**Comment by uvwxy** [comment]
Does this improve buildtime on github? Or just locally? My expectation would be this only improves local builds. Would be positively surprise if this also would speed CI builds.

Lgtm though :)

**Comment by RuffaloLavoisier** [comment]
@uvwxy 
Expectations are that CI build speeds will be improved.

**Comment by simonmicro** [comment]
I think this will also improve build times on GitHub... BUT:

- As far as I recall the `pio` command already uses more than one thread... Right? What is the default of `-j`? At least on my PC it is already blazingly fast...
- To retrieve the cpu thread count, please take a look [here](https://stackoverflow.com/a/68285053/11664229) - so we are not additional platform-depended on `nproc`...

**Comment by RuffaloLavoisier** [comment]
I added the platformio compile job because the way I was doing plan it doesn't seem to work with Github. (Lets find out more...) Now it seems that the time to build all models will be shorter than before.

**Comment by RuffaloLavoisier** [comment]
> * my PC it is already blazingly fast

:+1: 
@simonmicro 
But this is the Github CI system.

**Comment by RuffaloLavoisier** [comment]
>What is the default of `-j`?

Control a number of parallel build jobs. Default is a number of CPUs in a system.
See https://docs.platformio.org/en/latest//core/userguide/cmd_run.html#cmdoption-pio-run-j

wait...does this mean the number of CPUs? Does that mean parallel work?

File: .github/buildEverything.py [hunk 1]
```diff
@@ -72,16 +73,19 @@ def doBuild(makeDebug):
             configOut.close()
 
             # Compile firmware
-            logging.info('Compiling ' + filename + '...')
+            compile_start_time = datetime.now().strftime("%m/%d-%H:%M:%S")
+            logging.info('Compiling ' + compile_start_time + ' - ' + filename + '...')
+            job_number = str(os.cpu_count() * 2)
```
Comment (simonmicro) [inline .github/buildEverything.py:78]: The hyper-threads are already counted into that number!

**Review by simonmicro — CHANGES_REQUESTED** [review]

**Comment by simonmicro** [comment]
> Default is a number of CPUs in a system

And I don't see any advantage of explicitly setting the CPU core count using `-j`...

**Comment by RuffaloLavoisier** [comment]
Eureka... I think I found a way. I'll open up PR again.

**Comment by RuffaloLavoisier** [comment]
See #221

**Comment by RuffaloLavoisier** [comment]
In my humble opnion, I recommend working with one history or article. That way, we can check the previous changes and issues so you don're not confused. This PR has been disabled because a new PR was accidentally opened. I apologize.

**Pull request closed without merging**